modules/ai_engine_embedding/src/Service/EntityUpdate.php

```diff
@@ -330,7 +332,29 @@ protected function isIndexable(EntityInterface $entity) {
    *   TRUE if the entity should be embedded, FALSE otherwise.
    */
   protected function isSupportedEntityType(EntityInterface $entity) {
-    return $entity->getEntityTypeId() === 'node';
+    $entity_type_id = $entity->getEntityTypeId();
+
+    if ($entity_type_id == 'media') {
+      return $this->isSupportedMediaType($entity);
+    }
+    else {
+      return in_array($entity->getEntityTypeId(), self::ALLOWED_ENTITIES);
+    }
+  }
+
+  /**
+   * Checks if an entity is supported by the embedding system.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity to check.
+   *
+   * @return bool
+   *   TRUE if the entity should be embedded, FALSE otherwise.
+   */
+  protected function isSupportedMediaType(EntityInterface $entity) {
+    $config = $this->configFactory->get('ai_engine_embedding.settings');
+    $allowed_media_types = $config->get('included_media_types');
+    return in_array($entity->bundle(), $allowed_media_types);
```

Comment:
I'm guessing that this will throw an error on sites that already have this module enabled but did not resave the form.  Can not check in_array on NULL.

Reply:
Ah good catch; I've added `?? []` for $allowed_media_types now.